umbraco/Umbraco-CMS
# V8: Added babel-plugin-angularjs-annotate to the tool chain for JS

This fixes #4901 and future proofs the js tool chain setup so we can better minify the JS and support ES6 syntax better.

To test:

In windows explorer navigate to /src/Umbraco.Web.UI/umbraco/js/ and make a note of the umbraco.directives.js file size (assuming its there from a previous build, if not build the js first)

In the /src/Umbraco.Web.UI.Client folder run `npm install` to pull down the new package.

Then run `gulp js`. Then switch back to windows explore to see if the file size increases, we are expecting it to as this mod adds in an array at the front of all our DI references in our directives and controllers etc. Its this array that makes it safe for us to minify the files with more :)

Hit up the back office and double check everything is "working" as it should. As all the JS is loaded when you login if there was a JS error it should appear straight away in the dev console or the back office would simply not work/render. It should all just work as is.

Its possible that you might want to sit on this PR and not push it in until needed but I think its better to have it in now as its another bit of the puzzle for getting us to ES6 and smaller file sizes. Better to have it in I think.
